sjb/syntax.md

```diff
@@ -0,0 +1,211 @@
+# `sjb` YAML Syntax Reference
+
+A job configuration using `sjb` will take one of two forms: a stand-alone job
+or one that inherits from another. A standalone job can declare options; a job
+with a parent can extend or override any option from the parent.
+
+Parent-child relationships can nest -- to generate a job, the topmost parent
+configuration is loaded, then iteratively extended and overridden by children
+configurations until the bottommost configuration has been evaluated.
+
+## `parameters`
+
+`parameters` is an optional field which adds parameters to the Jenkins job. The
+field contains a list of parameter definitions. The parameter definition syntax
+is:
+
+```yaml
+parameters:
+  - name: ""          # environment variable name
+    description: ""   # human-readable description
+    default_value: "" # default value [optional]
+```
+
+## `provision`
+
+`provision` is a required field which gives the options for provisioning a VM
+in AWS EC2 for the job. The syntax is:
+
+```yaml
+provision:
+  os: ""       # operating system. only 'rhel' is supported
+  stage: ""    # image stage. one of ['bare', 'base', 'build', 'install']
+  provider: "" # cloud provider. only 'aws' is supported
+```
+
+## `sync_repos`
+
+`sync_repos` is an optional field which declares which repositories are to be
+synced and how they are to be synced. This field will add build stages to sync
+the repository as well as update the job description with details about what
+repos the job instance is being run with. The syntax is:
+
+```yaml
+sync_repos:
+  - name: "" # repository name under github.com/openshift
+    type: "" # 'pull_request' for a PR sync [optional]
```

Comment:
If `type` is not set it defaults to syncing the branch, if `type` is set to `"pull_request"` it will sync the branch and then merge the PR into that branch.